src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -2535,6 +2550,21 @@ private CancellableOutputStream putOrReference(
     }
   }
 
+  private void deleteExpiredKey(String expiredKey) throws IOException {
+    Path path = getPath(expiredKey);
+    if (configs.getWorker().getStorages().get(0).isPublishTtlMetric()) {
+      publishExpirationMetric(path);
+    }
+    Files.delete(path);
```

Comment:
Reverse these. I don't want to delay the file deletion by your publish, and at the time you publish here, we haven't deleted yet...

Reply:
fixed